stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/ChatDomainImpl.kt

```diff
@@ -356,7 +356,9 @@ internal class ChatDomainImpl internal constructor(
     private fun startListening() {
         if (eventSubscription.isDisposed) {
             eventSubscription = client.subscribe {
-                eventHandler.handleEvents(listOf(it))
+                scope.launch {
```

Comment:
We could possibly avoid using Flows as synchronization mechanisms if we made subscriptions suspending functions which would allow us to subscribe inside of a single coroutine body.

However I do not know if this may come at a cost for our users who do not use Coroutines. (would need further evaluation)